packages/shared/src/modules/users/libs/validation-schemas/user-sign-in.validation-schema.ts

```diff
@@ -0,0 +1,20 @@
+import { z, type ZodType } from "zod";
+
+import { UserValidationMessage, UserValidationRule } from "../enums/enums.js";
+import { type UserSignInRequestDto } from "../types/types.js";
+
+const userSignIn: ZodType<UserSignInRequestDto> = z.strictObject({
```

Comment:
We do not need to type this object explicitly, it will infer type from the library correctly. Currently it overwrites the type which is ZodObject, and not ZodType, and in the future it can lead to the mistakes as it's typed incorrect here

```suggestion
const userSignIn = z.strictObject({
```

Reply:
I know that there is same problem with userSignUp schema, I will ask in related PR to sign up flow to change this as well